### PR TITLE
View/charactersheet

### PR DESCRIPTION
**Pull request by Hich2000**
created the charactersheet layout as discussed. if changes are needed only App.vue needs to be changed with proper colspans and rowspans and it should cascade down.

### CODE REVIEW

File: src/views/traits/trait-input.vue [hunk 1]
```diff
@@ -6,10 +6,14 @@ import { state } from '@/singletons/state.js'
...
 
 const props = defineProps({
-  traitName: String
+  traitName: String,
+  colspanLabel: Number,
+  colspanInput: Number,
 })
 
 const traitName = props.traitName as TraitKey
+const colspanLabel = 'col-span-' + props.colspanLabel;
```
Comment (BanterTheGreat) [inline src/views/traits/trait-input.vue]: Use string interpolation

File: src/views/traits/trait-input.vue [hunk 2]
```diff
@@ -22,10 +26,10 @@ function updateTrait(event: any) {
 </script>
 
 <template>
-  <div>
-    <label>
-      {{ traitName }}:
-      <input id="trait-identity" v-model="TraitInput" @change="updateTrait" />
-    </label>
+  <div class="text-center font-bold uppercase" :class=colspanLabel>
+    {{ traitName }}
```
Comment (BanterTheGreat) [inline src/views/traits/trait-input.vue:30]: It feels kind of weird that we are using a Keyof to display text in a segment. But It'll do for now. Doubt the model will change and break this.

**Review by BanterTheGreat — APPROVED** [review]

File: src/views/traits/trait-input.vue (hunk 2 above)
Comment (BanterTheGreat) [reply src/views/traits/trait-input.vue:30]: This reminds me, why are we typing the prop as a string, to then cast it to a keyof. Why not just type the prop directly?

Comment (Hich2000) [reply src/views/traits/trait-input.vue:30]: was the first thing I made and didn't know how to do it by passing the type directly. might fix this specifically later

**Review by BanterTheGreat — APPROVED** [review]

**Pull request merged**